lua/dingllm.lua

```diff
@@ -5,6 +5,21 @@ local function get_api_key(name)
   return os.getenv(name)
 end
 
+local function get_file_extension()
+  local filename = vim.fn.expand("%:t")
+  local ext = filename:match("^.+(%..+)$")
+
+  return ext
+end
+
+function M.create_prompt(prompt)
+  local ext = get_file_extension()
+  if ext then
+    prompt = prompt:gsub("$file_ext", ext)
+  end
+  return prompt
+end
+
```

Comment:
Sorry for the back and forth and also lag. But I meant actually make system prompt a variable that can be either a function or string. If it's a function, take whatever it returns. If it's a string, use as is. If that makes sense!